### ISSUE TITLE
Show subset of countries in custom order

### ISSUE THREAD

**Issue by manuelmeurer** [stefanpenner/country_select#194]
Firstly, thanks for this great gem! 😄 

I'd like to show only a few countries (using the `only` option), but in custom order, not sorted alphabetically.
There doesn't seem to be any way to do this currently, correct?
[Here](https://github.com/stefanpenner/country_select/blob/4ea18b975e5facef354d0e5851afb2eb937dfcdb/lib/country_select/tag_helper.rb#L69) the order of the country codes passed in via the `only` option is reset, but even when I tried to switch those variables, the countries are then sorted in `country_options_for`.

A workaround I'm using right now is to call `country_select` with `only: %w(dummy), priority_countries: %w(DE AT CH)`.
The priority countries are then displayed in the order I want them in, and the `only` value is disregarded, since it doesn't contain valid country codes (but it can't be `nil` or empty either, then all countries are displayed).

Long story short - is this a feature you would be interested in / would accept a PR for? 😸

**Comment by scudco** [comment]
The ability to do this makes sense to me, but I'm still trying to work through how this would best be explained in the README. Perhaps we could add an option that follows the newer [Enumerable#in_order_of](https://api.rubyonrails.org/classes/Enumerable.html#method-i-in_order_of) rules. It has answers for some edge cases like when objects in the enumerable are not present in the order array (it removes them).

### 1. Option with `Enumerable#in_order_of` syntax
```ruby
country_select :user, :country_code, only: %w(CH AT DE), in_order_of: [:code, %w(DE AT CH)]
```

### 2. An option we add
```ruby
country_select :user, :country_code, only: %w(CH AT DE), order: %w(DE AT CH)
```

### 3. Simplest option
```ruby
country_select :user, :country_code, only: %w(DE AT CH) # Simply respects order of the `only` option
```

I believe we'd be able to utilize the first option directly on the internal countries array with Rails >= 7 and provide a backfilled implementation for Rails < 7.

The second option seems less verbose since the recommendation to the developer has always been to use ISO3166 codes in the options. Not sure if there'd ever be a need to use `in_order_of` with a different `Country` attribute.

The third option just means building the array as-is without sorting. This could present some issues if that `only` list is derived from some other attribute (e.g., `user.visited_countries`, `order.shippable_countries`) that may not contain a meaningful sort order. However, I imagine this could be just as easily remedied by explicitly sorting that array as it's passed into the `country_select` helper 🤔 It would just represent a departure from things being automatically sorted by name according to the app's locale.

@manuelmeurer Which option, if any, do you wish you had had? Maybe something else entirely? I'd definitely welcome a PR to allow custom sort orders.

**Comment by manuelmeurer** [comment]
@scudco Thanks for taking the time to think through this! :)

I think option 3 would make the most sense, especially since the `priority_countries` are already displayed in the order they are passed it, so it's only logical this applies to the `only` options as well IMO.
Maybe we can add a `sort` option that defaults to false, which would sort the `only` and `priority_countries` values if it's true?

Since this is a breaking change, it should definitely be released in a new major version.
